### ISSUE TITLE
Multi-output regression support for Transformer models

### ISSUE THREAD

**Issue by mangalgishreyas** [huggingface/transformers#4841]
# 🚀 Feature request

## Motivation

I am trying to build a shipping address to geo-code predictor using RoBERTa. Here the shipping address would be the text input and the output would be a gecode (latitude and longitude).
I tried using _robertaforsequenceclassification_ but it mentions that when the final layer consists of more than one class, a cross entropy loss would be used automatically. But I want to perform regression using the RMSE loss. It would be great if we can add the multi-output regression feature to the existing sequence classification pipeline.

**Comment by LysandreJik** [comment]
Hello! The cross entropy loss is only used if you provide the `labels` for the model to compute the loss. If you don't provide the labels, the model doesn't output any loss, only the logits!

You can then use these logits with your labels and the loss of your choosing.
